proto/text.go

```diff
@@ -856,6 +856,22 @@ type TextMarshaler struct {
 	ExpandAny bool // expand google.protobuf.Any messages of known types
 }
 
+var bufWriterPool = sync.Pool{
```

Comment:
This bufWriterPool is a separate optimization.
I think we can try to contribute this to golang/protobuf and then merge this into the gogo/protobuf fork, when merged into golang/protobuf.